### PR TITLE
fix(blade): bottomsheet isOpen state, simplify isOpen logic & glue code

### PR DESCRIPTION
**Pull request by anuraghazra**
Changes: 

- Fixes isOpen state bug in BottomSheet
- Simplifies the glue code between dropdown & bottomsheet

Video walkthrough: https://www.loom.com/share/f24fcb51b245431fbf1a0aeb53cea287

### CODE REVIEW

File: packages/blade/src/components/BottomSheet/BottomSheet.web.tsx [hunk 1]
```diff
@@ -171,56 +170,42 @@ const _BottomSheet = ({
...
+  const close = React.useCallback(() => {
+    onDismiss?.();
+    bottomSheetAndDropdownGlue?.onBottomSheetDismiss();
+  }, [bottomSheetAndDropdownGlue, onDismiss]);
 
   // sync controlled state to our actions
   React.useEffect(() => {
-    if (isOpen === true) {
-      open();
+    if (_isOpen === true) {
```
Comment (divyanshu013) [inline packages/blade/src/components/BottomSheet/BottomSheet.web.tsx:193]: Don't need to compare against booleans: `if (_isOpen) ... else`

File: packages/blade/src/components/BottomSheet/BottomSheet.web.tsx [hunk 2]
```diff
@@ -171,56 +170,42 @@ const _BottomSheet = ({
     }
   }, [initialFocusRef]);
 
-  const close = React.useCallback(() => {
+  const handleOnOpen = React.useCallback(() => {
+    setPositionY(dimensions.height * 0.5);
+    scrollLockRef.current.activate();
+    originalFocusElement.current =
+      originalFocusElement.current ?? (document.activeElement as HTMLElement);
+    focusOnInitialRef();
```
[the remark this answers is not in the document]
Comment (anuraghazra) [reply packages/blade/src/components/BottomSheet/BottomSheet.web.tsx:178]: Added

File: .changeset/soft-bees-pay.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+"@razorpay/blade": patch
+---
+
+fix(blade): bottomsheet isOpen state, simplify isOpen logic & glue code
```
[the remark this answers is not in the document]
Comment (anuraghazra) [reply .changeset/soft-bees-pay.md:5]: Done.

File: .changeset/soft-bees-pay.md [hunk 4]
```diff
@@ -3,3 +3,6 @@
 ---
 
 fix(blade): bottomsheet isOpen state, simplify isOpen logic & glue code
+
+Previously if users did not changed the isOpen state to false inside `onDismiss` the bottomsheet's internal state will still remain "open", but the bottomsheet would visually be hidden and the backdrop will still remain, this fixes the bug so that internally we won't modify the bottomsheet's position instead we will just call the `onDismiss`. [Check the loom](https://www.loom.com/share/f24fcb51b245431fbf1a0aeb53cea287) video here for more info.
```
Comment (divyanshu013) [inline .changeset/soft-bees-pay.md:7]: Bit too much details for changelog but okay 😅

File: packages/blade/src/components/BottomSheet/BottomSheet.stories.tsx [hunk 5]
```diff
@@ -404,14 +404,15 @@ export const WithSelect = BottomSheetWithSelectTemplate.bind({});
...
-  const [isOpen, setIsOpen] = React.useState(true);
+  const [isOpen, setIsOpen] = React.useState(false);
 
   return (
     <BaseBox>
       <Button onClick={() => setIsOpen(true)}>Open</Button>
       <BottomSheetComponent
         isOpen={isOpen}
         onDismiss={() => {
+          console.log('dismiss');
```
Comment (divyanshu013) [inline packages/blade/src/components/BottomSheet/BottomSheet.stories.tsx:415]: Stray log

File: packages/blade/src/components/BottomSheet/BottomSheet.stories.tsx [hunk 6]
```diff
@@ -404,14 +404,15 @@ export const WithSelect = BottomSheetWithSelectTemplate.bind({});
 WithSelect.storyName = 'WithSelect';
 
 const BottomSheetScrollingContentTemplate: ComponentStory<typeof BottomSheetComponent> = () => {
-  const [isOpen, setIsOpen] = React.useState(true);
+  const [isOpen, setIsOpen] = React.useState(false);
```
Comment (divyanshu013) [inline packages/blade/src/components/BottomSheet/BottomSheet.stories.tsx]: Do you mean to change this?

File: .changeset/soft-bees-pay.md (hunk 4 above)
Comment (anuraghazra) [reply .changeset/soft-bees-pay.md:7]: lol yeah, 

"Okay chatgpt, tldr this paragraph."

File: packages/blade/src/components/BottomSheet/BottomSheet.stories.tsx (hunk 6 above)
Comment (anuraghazra) [reply packages/blade/src/components/BottomSheet/BottomSheet.stories.tsx]: Reverted

Comment (anuraghazra) [reply packages/blade/src/components/BottomSheet/BottomSheet.stories.tsx]: Let's keep it false only, set it to true for debugging purposes.

File: packages/blade/src/components/BottomSheet/BottomSheet.stories.tsx (hunk 5 above)
Comment (anuraghazra) [reply packages/blade/src/components/BottomSheet/BottomSheet.stories.tsx:415]: removed.

**Review by divyanshu013 — APPROVED** [review]
Looks good to me.

**Pull request merged**